WordPress/Classes/ViewRelated/Ratings/AppFeedbackPromptView.swift

```diff
@@ -81,30 +80,44 @@ class AppFeedbackPromptView: UIView {
         rightButton.addTarget(self, action: #selector(self.rightButtonTouched), for: .touchUpInside)
         buttonStack.addArrangedSubview(rightButton)
 
-        setNeedsUpdateConstraints()
-    }
-
-    override func updateConstraints() {
         setupConstraints()
-        super.updateConstraints()
     }
 
-    func setupConstraints() {
-        guard constraintsAdded == false else {
-            return
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        self.buttonStack.axis = .horizontal
+        self.buttonStack.isLayoutMarginsRelativeArrangement = true
+
+        // measure the width of the view with the new font sizes to see if the buttons are too wide.
+        leftButton.updateFontSizeToMatchSystem()
+        rightButton.updateFontSizeToMatchSystem()
+        let newLayoutSize = systemLayoutSizeFitting(UILayoutFittingCompressedSize)
+
+        // if the new width is too wide, change the axis of the stack view
+        if newLayoutSize.width > UIScreen.main.bounds.size.width {
```

Comment:
Should we compare to our superview's width instead? Just thinking on the iPad, the notifications list is displayed in a narrow pane, but the screen itself is obviously much wider.

Reply:
Just tested this – at default font size on iPad, the right button gets truncated:

<img width="329" alt="screen shot 2018-05-10 at 12 14 47" src="https://user-images.githubusercontent.com/4780/39867234-d9e03b12-544b-11e8-9cb8-76049fad9089.png">

(although it'd be nice if we could make those fit horizontally on iPad)

Reply:
Ah, nice catch on iPad. You're right, we should probably compare to superview. 👍